sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuJsonTuple.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+// import ai.rapids.cudf.{ColumnVector, Table}
+import ai.rapids.cudf.{Scalar, Table}
+import com.nvidia.spark.rapids.GpuExpressionsUtils.columnarEvalToColumn
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import com.nvidia.spark.rapids.shims.ShimExpression
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions.{Expression, NamedExpression}
+import org.apache.spark.sql.types.{DataType, StringType, StructField, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class GpuJsonTuple(children: Seq[Expression]) extends GpuGenerator 
+  with ShimExpression {
+  override def nullable: Boolean = false // a row is always returned
+  @transient private lazy val jsonExpr: Expression = children.head
+  @transient private lazy val fieldExpressions: Seq[Expression] = children.tail
+
+  override def elementSchema: StructType = StructType(fieldExpressions.zipWithIndex.map {
+    case (_, idx) => StructField(s"c$idx", StringType, nullable = true)
+  })
+
+  override def prettyName: String = "json_tuple"
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (children.length < 2) {
+      TypeCheckResult.TypeCheckFailure(
+        s"$prettyName has wrong number of auguments: expected > 1, but found ${children.length}"
+      )
+    } else if (children.forall(child => child.dataType == StringType)) {
+      TypeCheckResult.TypeCheckSuccess
+    } else {
+      TypeCheckResult.TypeCheckFailure(s"$prettyName only supports string type input")
+    }
+  }
+
+  def generate(inputBatch: ColumnarBatch,
+      generatorOffset: Int,
+      outer: Boolean): ColumnarBatch = {
+
+    val jsonGpuExpr = GpuBindReferences.bindReference(jsonExpr, 
+      Seq(jsonExpr.asInstanceOf[NamedExpression].toAttribute))
+    val schema = Array.fill[DataType](fieldExpressions.length)(StringType)
+
+    withResource(columnarEvalToColumn(jsonGpuExpr, inputBatch).getBase) { json =>
+      val fieldCols = fieldExpressions.safeMap(field => columnarEvalToColumn(field, inputBatch))
```

Comment:
If we know that the inputs are scalar values, why do we convert them into columns and then pull the scalar value back out again?

In these cases you would want to call something like 
```
val resColumns = fieldExpressions.safeMap { field =>
  withResourceIfAllowed(field.columnarEval(batch)) { fieldVal =>
    val fieldScalarPattern = fieldVal match {
      case fieldScalar: GpuScalar =>  
        // TODO need some escapes here in case the name has a $ or a . or something like that in it.
        Scalar.fromString("$." + fieldScalar.getJavaString)
      case _ => throw new ... "Not expected data type only scalars are supported"
    }
    json.getJSONObject(fieldScalarPattern)
  }
}
withResource(resColumns) { _ =>
  withResource(pathScalars.safeMap(json.getJSONObject(_))) { resCols =>
    withResource(new Table(resCols: _*)) { tbl =>
      GpuColumnVector.from(tbl, schema)
    }
  }
}
```

Reply:
Updated. Thank you very much!